packages/flutter_tools/lib/src/commands/build_aar.dart

```diff
@@ -104,15 +106,18 @@ class BuildAarCommand extends BuildSubCommand {
 
     for (final String buildMode in const <String>['debug', 'profile', 'release']) {
       if (boolArg(buildMode)) {
-        androidBuildInfo.add(AndroidBuildInfo(
-          BuildInfo(BuildMode.fromName(buildMode), stringArg('flavor'), treeShakeIcons: boolArg('tree-shake-icons')),
-          targetArchs: targetArchitectures,
-        ));
+        androidBuildInfo.add(
+          AndroidBuildInfo(
+            getBuildInfo(forcedBuildMode: BuildMode.fromName(buildMode)),
```

Comment:
https://github.com/flutter/flutter/issues/56068#issuecomment-622054920
> The build_aar command would need to be refactored to get the buildInfo from the flutter command instead of manually constructing it.

Is this what you meant, @jonahwilliams?

Reply:
Yup! though there are some edges here. For example, If a user builds a debug aar with obfuscation enabled, the sources might be scrambled. This might cause confusion if there is some expectation that these flags only work on "release" builds

Reply:
@xster What do you think the best user experience would be here?  
1. Ignore the flags for the debug and profile output
1. Ignore the flags for the debug and profile output, and print a warning
1. Only allow the flags accompanied by `--no-debug` and `--no-profile`

Reply:
FWIW  we do support --obfuscate on --debug `flutter build apk --debug` today, but not through flutter run. The former isn't very common so I didn't think it merited any interception, but it might be more common to build artifacts together here

Reply:
I vote for 2

Reply:
```
$ flutter build aar --target-platform=android-arm --obfuscate --split-debug-info=foo/                                  
Building flutter tool...
Changing current working directory to: /Users/magder/Projects/aaa
Dart obfuscation is not supported in Debug mode, skipping.              
Running Gradle task 'assembleAarDebug'...                               
Running Gradle task 'assembleAarDebug'... Done                     20.6s
✓ Built build/host/outputs/repo.
Dart obfuscation is not supported in Profile mode, skipping.            
Running Gradle task 'assembleAarProfile'...                             
Running Gradle task 'assembleAarProfile'... Done                   21.4s
✓ Built build/host/outputs/repo.
Running Gradle task 'assembleAarRelease'...                             
Running Gradle task 'assembleAarRelease'... Done                    3.2s
✓ Built build/host/outputs/repo.
```

Reply:
LG. "skipping" is a bit ambiguous since you didn't skip building debug/profile. Maybe just say building debug/release in unobfuscated mode?

Reply:
Changed to: "Dart obfuscation is not supported in Debug mode, building as unobfuscated."